ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezSessionState.java

```diff
@@ -109,7 +112,7 @@
 
   private final HiveConf conf;
   private Path tezScratchDir;
-  private LocalResource appJarLr;
+  private Collection<LocalResource> appJarLrs;
```

Comment:
I think instead of making it a "multi file" stuff; we can just add it to the shader in ql/pom.xml

Reply:
Thanks for the suggestion. Unfortunately it seems that the generated HiveParser.class has some huge methods, which causes the shade plugin to fail.

Error creating shaded jar: Problem shading JAR /home/antals/.m2/repository/org/apache/hive/hive-parser/4.0.0-SNAPSHOT/hive-parser-4.0.0-SNAPSHOT.jar entry org/apache/hadoop/hive/ql/parse/HiveParser.class: java.lang.RuntimeException: Method code too large!

Reply:
HIVE-23857 fixed the Method code too large! issue.